### PR TITLE
Disable Multichain, support Lifi, remove Curve

### PR DESCRIPTION
**Pull request by af-afk**

### CODE REVIEW

File: tests/integrations/ethereum/util/mock_rpc.go [hunk 1]
```diff
@@ -109,15 +117,16 @@ func MockRpcClient(rpcMethods_ map[string]interface{}, callMethods_ map[string]m
...
+		if method := request.Method; method != "eth_call" {
 			expectedResponse := rpcMethods[request.Method]
 
 			if expectedResponse != nil {
 				response.Result = expectedResponse
 				marshalResponse(response, rw)
 				return
 			} else {
-				marshalError(response, rw, "Unsupported method")
+				msg := fmt.Sprintf("Unsupported method, was %v, need eth_call", method)
```
Comment (eli-d) [inline tests/integrations/ethereum/util/mock_rpc.go]: this comment is inaccurate - the method can be either `eth_call` or anything in `rpcMethods`

File: tests/integrations/ethereum/util/mock_rpc.go [hunk 2]
```diff
@@ -38,7 +38,15 @@ func methodSigHash(sig string) string {
...
 
 func responseErr(response serverResponse, err interface{}) serverResponse {
-	response.Error = err
+	response.Error = struct {
+		Code    int         `json:"code"`
+		Message string      `json:"message"`
+		Data    interface{} `json:"data,omitempty"`
+	}{
+		Code: 999,
+		Message: fmt.Sprintf("responding with err: %v", err),
```
Comment (eli-d) [inline tests/integrations/ethereum/util/mock_rpc.go:47]: seems redundant to have a message that's the exact same as data, when this struct is used a single time

File: common/ethereum/applications/lifi/lifi.go [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+			)
+		}
+
+		return swappedGeneric(
+			fluidTokenContract,
+			tokenDecimals,
+			unpacked,
+		)
+
+	case lifiAssetSwappedTopic:
```
Comment (eli-d) [inline common/ethereum/applications/lifi/lifi.go:507]: will this be emitted from the same addresses as the other topics (therefore crashing the app server?) or is it from a separate contract? or is this meant to use `assetSwapped`?

File: common/ethereum/applications/lifi/lifi.go [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+
+// genericSwapCompleted decodes "LiFiGenericSwapCompleted" and is
+// determined by testing which side of the swap is a Fluid Asset, then
+// setting the volume based on the volume for that side of the trade.
+// Assumes 7 fields in the event decoding. Ie, if I swap 10 USDC for 9
+// fUSDC, then the volume is 9.
+func genericSwapCompleted(fluidTokenContract ethCommon.Address, tokenDecimals int, unpacked []interface{}) (feeData applications.ApplicationFeeData, err error) {
+	if l := len(unpacked); l != 7 {
+		return feeData, fmt.Errorf(
+			"unpacked the wrong number of values for LiFiGenericSwapCompleted! Expected 8, got %v",
```
Comment (eli-d) [inline common/ethereum/applications/lifi/lifi.go]: should say expected 7

File: common/ethereum/applications/lifi/lifi.go [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+
+	return feeData, nil
+}
+
+// swappedGeneric decodes LiFiSwappedGeneric, following largely the same
+// behaviour as genericSwapCompleted. Expects 1 topic, and 6 unpacked items.
+func swappedGeneric(fluidTokenContract ethCommon.Address, tokenDecimals int, unpacked []interface{}) (feeData applications.ApplicationFeeData, err error) {
+	if l := len(unpacked); l != 6 {
+		return feeData, fmt.Errorf(
+			"unpacked the wrong number of values! Expected 8, got %v",
```
Comment (eli-d) [inline common/ethereum/applications/lifi/lifi.go]: should say 6

File: common/ethereum/applications/lifi/lifi.go [hunk 6]
```diff
@@ -0,0 +1,516 @@
...
+
+	return feeData, nil
+}
+
+// assetSwapped decodes events of AssetSwapped. Follows the same
+// behaviour as it's predecessors.
+func assetSwapped(fluidTokenContract ethCommon.Address, tokenDecimals int, unpacked []interface{}) (feeData applications.ApplicationFeeData, err error) {
+	if l := len(unpacked); l != 7 {
+		return feeData, fmt.Errorf(
+			"unpacked the wrong number of values! Expected 8, got %v",
```
Comment (eli-d) [inline common/ethereum/applications/lifi/lifi.go]: should say 7

File: tests/integrations/ethereum/util/mock_rpc.go (hunk 2 above)
Comment (af-afk) [reply tests/integrations/ethereum/util/mock_rpc.go:47]: Just having the string is incompatible with go-ethereum's internal rpc error type, which expects a structure like this. If an error happens and it isn't parsed, it errors out with a failure to decode it, ignoring the content.

File: common/ethereum/applications/lifi/lifi.go (hunk 3 above)
Comment (af-afk) [reply common/ethereum/applications/lifi/lifi.go:507]: It won't be emitted anyway. If it does, we need to support it with a test. I couldn't find a transaction online to use for a test except on a small Bitcoin shitcoin chain that didn't have an archive node available with the public RPC (the only event was from last year). It seems to be a non supported event that Lifi doesn't emit (though supported via the ABI - with my cursory search).

**Review by eli-d — APPROVED** [review]

**Pull request merged**